tests/src/providers/testqgseptprovider.cpp

```diff
@@ -641,6 +646,43 @@ void TestQgsEptProvider::testPointCloudIndex()
     QVERIFY( bounds.yMax() == 88000 );
     QVERIFY( bounds.zMax() == 0 );
   }
+  qDebug() << __PRETTY_FUNCTION__ << "finished";
+}
+
+#include "qgspointcloudstatscalculator.h"
+#include "qgsstatisticalsummary.h"
+#include "qgsfeedback.h"
+
+void TestQgsEptProvider::testStatsCalculator()
+{
+  std::unique_ptr< QgsPointCloudLayer > layer = std::make_unique< QgsPointCloudLayer >( mTestDataDir + QStringLiteral( "point_clouds/ept/extrabytes-dataset/ept.json" ), QStringLiteral( "layer" ), QStringLiteral( "ept" ), mNoStatsOption );
+  QgsPointCloudIndex *index = layer->dataProvider()->index();
+  QgsPointCloudStatsCalculator calculator( index );
+
+
+  QVector<QgsPointCloudAttribute> attributes = index->attributes().attributes();
+  for ( int i = 0; i < attributes.size(); ++i )
+  {
+    QString name = attributes.at( i ).name();
+    if ( name == QStringLiteral( "X" ) || name == QStringLiteral( "Y" ) || name == QStringLiteral( "Z" ) )
+    {
+      attributes.removeAt( i );
+      --i;
+    }
+  }
```

Comment:
please replace this with a hard-coded list of attributes that we want to calculate (instead of taking all attributes and removing X+Y+Z) ... logic with conditions and loops is better to be avoided in unit tests to lower the chance that the test itself is broken (or gets broken in future)